src/protocol/tx-validity.md

```diff
@@ -205,13 +284,45 @@ Once the free balances are computed, the [script is executed](../fuel-vm/index.m
 1. The unspent free balance `unspentBalance` for each asset ID.
 1. The unspent gas `unspentGas` from the `$ggas` register.
 
-The fees incurred for a transaction are `ceiling(((size(tx) * GAS_PER_BYTE) + (tx.scriptGasLimit - unspentGas) + sum(tx.inputs[i].predicateGasUsed)) * tx.gasPrice / GAS_PRICE_FACTOR)`.
-
-`size(tx)` includes the entire transaction serialized according to the transaction format, including witness data.
+`size(tx)` encompasses the entire transaction serialized according to the transaction format, including witness data.
 This ensures every byte of block space either on Fuel or corresponding DA layer can be accounted for.
 
 If the transaction as included in a block does not match this final transaction, the block is invalid.
 
+### Fees
+
+The cost of a transaction can be described by:
+
+```py
+cost(tx) = gas_to_fee(gas_cost(tx) - unspentGas, tx.gasPrice)
+```
+
+where:
+
+- `gas_cost(tx)` is the final cost of the transaction in gas, including gas fees incurred from:
+  - Intrinsic fees:
+    - The number of bytes comprising the transaction
+    - Processing inputs and outputs
+    - VM initialization
+  - Predicate and script execution
+- `unspentGas` is the amount gas left over after intrinsic fees and execution of the transaction, extracted from the `$ggas` register. Converting unspent gas to a fee describes how much "change" is left over from the user's payment; the block producer collects this unspent gas as reward.
+- `gas_to_fee` is a function that converts gas to a concrete fee based on a given gas price.
+
+Fees incurred by transaction processing outside the context of execution are collectively referred to as intrinsic fees. Intrinsic fees include the cost of storing the transaction, calculated on a per-byte basis, the cost of processing inputs and outputs, including signature verification, and initialization of the VM prior to any predicate or script execution. Because intrinsic fees are independent of execution, they can be determined _a priori_ and represent the bare minimum cost of the transaction.
+
+Users wishing to submit transactions can incentivize block producers to include their transactions by providing a higher reward in the form of more unspent gas. This is achieved by specifying a higher [gas limit](../tx-format/transaction.md) on the transaction.
+
+A naturally occurring result of a variable gas limit is the concept of minimum and maximum fees. The minimum fee is, thus, the exact fee required to pay the fee balance, while the maximum fee is the minimum fee plus the gas limit:
```

Comment:
It might be more clear to say that min_fee is the instrinsic cost of the tx, while max fee covers any additional execution fees for custom logic that is performed in terms of scripts and contracts.